app/code/community/HE/TwoFactorAuth/sql/he_twofactorauth/mysql4-upgrade-1.1.10-1.1.11.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/** @var Mage_Core_Model_Resource_Setup $this */
+$this->startSetup();
+$connection = $this->getConnection();
+
+$tableName = $this->getTable('he_twofactorauth/trusted');
+
+$table = $connection->newTable($tableName);
+$table->addColumn('trusted_id', Varien_Db_Ddl_Table::TYPE_INTEGER, 10, [
+    'unsigned' => true,
+    'nullable' => false,
+    'primary' => true,
+    'identity' => true,
+], 'Trusted Device ID');
+$table->addColumn('date_time', Varien_Db_Ddl_Table::TYPE_TIMESTAMP, null, [
+    'nullable' => false,
+    'default' => '0000-00-00 00:00:00',
+], 'Creation time');
+$table->addColumn('user_id', Varien_Db_Ddl_Table::TYPE_INTEGER, 10, [
+    'unsigned' => true,
+    'nullable' => false,
+], 'Admin user ID');
+$table->addColumn('device_name', Varien_Db_Ddl_Table::TYPE_TEXT, 255, [
+    'nullable' => false,
+], 'Device Name');
+$table->addColumn('token', Varien_Db_Ddl_Table::TYPE_TEXT, 255, [
+    'nullable' => false,
+], 'Token');
+$table->addColumn('last_ip', Varien_Db_Ddl_Table::TYPE_TEXT, 16, [
```

Comment:
nit: last_ip isn't really the "last" IP as much as this IP, since it's unique in this way.  I think it's fine, though maybe ip is better.

That said, I think the length should be 39 (for ipv6.)  I know most of our servers are ipv4 only but don't want to be surprised.